api/v1alpha4/common_types.go

```diff
@@ -79,6 +79,14 @@ const (
 
 	// InterruptibleLabel is the label used to mark the nodes that run on interruptible instances
 	InterruptibleLabel = "cluster.x-k8s.io/interruptible"
+
+	// ExternallyManagedAnnotation is an annotation that can be applied to InfraCluster resources to signify that
+	// some external system is managing the cluster infrastructure.
+	//
+	// Provider InfraCluster controllers will ignore resources with this annotation.
```

Comment:
should this assumption be documented in the provider contract?

Reply:
Just to confirm, that means adding it to https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/cluster-infrastructure.md? I've updated the diagram there and added a note to the top of the normal resource flow bit.

Is it worth adding a more detailed section about the externally managed infrastructure in here, or does that belong somewhere else in the docs?

Reply:
can you also add it to https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md with required changes for providers?

Reply:
@CecileRobertMichon I've added a small section to the bottom of the document. Let me know what you think, I am happy to add more context if you think it's required

Reply:
looks great, thanks